src/Async.hs

```diff
@@ -13,7 +15,7 @@ removeOne p (x:xs)
   | otherwise = (x :) <$> removeOne p xs
 
 waitAtLeast :: Int -> [Async a] -> IO [a]
-waitAtLeast 0 actions = mapM_ cancel actions >> pure []
+waitAtLeast 0 actions = traverse_ cancel actions $> []
```

Comment:
`>>` is another example of a function that is still in `Monad` for compatibility reasons. `*>` from `Applicative` should be used instead. In this case we can do even better with `$>` from `Functor` which is really since it allows us to do way with the `pure`.